sys/userlib/src/kipc.rs

```diff
@@ -3,6 +3,16 @@
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
 //! Operations implemented by IPC with the kernel task.
+//!
+//! # On checking return values
+//!
+//! All the functions in this module send IPCs to the kernel directly. It's not
+//! generally useful for us to check the return codes, except in cases where the
+//! IPC is defined as able to fail. We have no choice but to trust the kernel,
+//! since it controls everything.
+//!
+//! As a result, asserting on return codes when they can only be zero just
+//! wastes flash space in the supervisor.
```

Comment:
considered pedantic: perhaps 
```suggestion
//! As a result, asserting on return codes and lengths when they can only be zero just
//! wastes flash space in the supervisor.
```
or similar, given that we've also removed some assertions about response lengths?